dax/api.go

```diff
@@ -291,20 +292,60 @@ func (d *Dax) BatchGetItemPagesWithContext(aws.Context, *dynamodb.BatchGetItemIn
 	return d.unImpl()
 }
 
-func (d *Dax) QueryPages(*dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool) error {
-	return d.unImpl()
+func (d *Dax) QueryPages(input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool) error {
+	return d.QueryPagesWithContext(context.Background(), input, fn)
 }
 
-func (d *Dax) QueryPagesWithContext(aws.Context, *dynamodb.QueryInput, func(*dynamodb.QueryOutput, bool) bool, ...request.Option) error {
-	return d.unImpl()
+func (d *Dax) QueryPagesWithContext(ctx aws.Context, input *dynamodb.QueryInput, fn func(*dynamodb.QueryOutput, bool) bool, opts ...request.Option) error {
+	out, err := d.QueryWithContext(ctx, input, opts...)
+	if err != nil {
+		return err
+	}
+
+	fn(out, out.LastEvaluatedKey == nil)
```

Comment:
Function should return here if `fn` results to `false`. Otherwise it unncessarly executes later steps even when `fn` retuens  `false` for the first time. 
Eg:
`fn :=func(page *dynamodb.QueryOutput, lastPage bool) bool { return false }`

And I would suggest to refactor the code to handle both first page and rest of the pages in a single loop.